### DISCUSSION TITLE
Simulation of patch antenna example

### DISCUSSION THREAD

**Discussion by wolfc01** [thliebig/openEMS-Project#41] [Q&A Usage]
Hi,

I try to simulate a patch antenna as defined in [emtalk](https://www.emtalk.com/mpacalc.php?er=29.3&h=7&h_units_list=hmm&fr=&Operation=Analyze&La=63.065502581809&L_units_list=Lmm&Wa=88.796370196781&W_units_list=Wmm&Rin=1363.5
), based on altering the example Simple_patch_antenna.py

I altered the parameters (size, substrate epsR etc), and tried to add a microstrip line for feeding the patch.

But I get stuck at `port = FDTD.AddMSLPort(1, patch, start, stop, "x", "y", priority=5, edges2grid='xy')` throwing an exception `Exception: At least 5 lines in propagation direction required!`

Could somebody alter the attached file such that it works?

Best regards
Carl.

[Simple_Patch_Antenna.zip](https://github.com/thliebig/openEMS-Project/files/10892251/Simple_Patch_Antenna.zip)

**Comment by thliebig** [comment]
Looking at this "Microstrip Patch Antenna Calculator" I must say that this feeding approach is a horrible idea... The impedance at the edge of a patch is very high and feeding there is a terrible idea... You can feed from the side with a MSL, but not like this ;)

**Reply by thliebig** [reply]
As for your question, you have to define the smooth mesh first. Also your directions are wrong and the excitation is 0.

Furthermore I think the "AddEdges2Grid" seems to not work at the moment... I have to check that

**Reply by thliebig** [reply]
The "AddEdges2Grid" should be fixed, so if you are on linux you should git pull and rebuild

**Reply by thliebig** [reply]
with older python this should still work as it was I think? But I'm not sure...

**Comment by wolfc01** [comment]
Hi Thorsten,

Thanks for your advice. I'm just learning on the openEMS tool by comparing
results between openEMS and such online calcs.

I'm working on a patch antenna for real usage. This antenna will be
submerged in demi-water. Are you willing to create an initial script for
such an antenna in your tool?

I find it difficult right now as too many openEMS parameters are not fully
clear to me.

Parameters like resonance frequency and dielectric constants and material
type are known.

Best regards
Carl.





Op zo 5 mrt. 2023 20:00 schreef Thorsten Liebig ***@***.***>:
